sql/moz-fx-data-shared-prod/subscription_platform_derived/subplat_flow_events_v1/query.sql

```diff
@@ -0,0 +1,84 @@
+WITH new_flow_events AS (
+  SELECT
+    `timestamp` AS log_timestamp,
+    COALESCE(event_time, `timestamp`) AS event_time,
+    event_type,
+    flow_id,
+    user_id AS mozilla_account_id_sha256,
+    oauth_client_id,
+    service AS oauth_client_name,
+    product_id,
+    plan_id,
+    entrypoint,
+    entrypoint_experiment,
+    entrypoint_variation,
+    utm_campaign,
+    utm_content,
+    utm_medium,
+    utm_source,
+    utm_term,
+    promotion_code,
+    country_code,
+    country,
+    `language`,
+    os_name,
+    os_version,
+    ua_browser,
+    ua_version,
+  FROM
+    `moz-fx-data-shared-prod.firefox_accounts.fxa_all_events`
+  WHERE
+    fxa_log IN ('content', 'auth', 'stdout')
+    AND flow_id IS NOT NULL
+    {% if is_init() %}
+      AND DATE(`timestamp`) < CURRENT_DATE()
+    {% else %}
+      AND DATE(`timestamp`) = @date
+    {% endif %}
+),
+services_metadata AS (
+  SELECT
+    ARRAY_AGG(DISTINCT subplat_oauth_client.id IGNORE NULLS) AS subplat_oauth_client_ids,
+    ARRAY_AGG(DISTINCT subplat_oauth_client.name IGNORE NULLS) AS subplat_oauth_client_names,
+    ARRAY_AGG(DISTINCT stripe_product_id IGNORE NULLS) AS stripe_product_ids,
+    ARRAY_AGG(DISTINCT stripe_plan_id IGNORE NULLS) AS stripe_plan_ids
+  FROM
+    `moz-fx-data-shared-prod.subscription_platform_derived.services_v1`
+  LEFT JOIN
+    UNNEST(subplat_oauth_clients) AS subplat_oauth_client
+  LEFT JOIN
+    UNNEST(stripe_product_ids) AS stripe_product_id
+  LEFT JOIN
+    UNNEST(stripe_plan_ids) AS stripe_plan_id
+),
+existing_flow_ids AS (
+  {% if is_init() %}
+    SELECT
+      CAST(NULL AS STRING) AS flow_id
+  {% else %}
+    SELECT DISTINCT
+      flow_id
+    FROM
+      `moz-fx-data-shared-prod.subscription_platform_derived.subplat_flow_events_v1`
+  {% endif %}
+)
+SELECT
+  new_flow_events.*
+FROM
+  new_flow_events
+CROSS JOIN
+  services_metadata
+LEFT JOIN
+  existing_flow_ids
+ON
+  new_flow_events.flow_id = existing_flow_ids.flow_id
+QUALIFY
+  LOGICAL_OR(
+    new_flow_events.oauth_client_id IN UNNEST(services_metadata.subplat_oauth_client_ids)
+    OR new_flow_events.oauth_client_name IN UNNEST(services_metadata.subplat_oauth_client_names)
+    -- For a while Bedrock incorrectly passed VPN's OAuth client name as the OAuth client ID.
+    OR new_flow_events.oauth_client_id IN UNNEST(services_metadata.subplat_oauth_client_names)
+    OR new_flow_events.product_id IN UNNEST(services_metadata.stripe_product_ids)
+    OR new_flow_events.plan_id IN UNNEST(services_metadata.stripe_plan_ids)
+  ) OVER (PARTITION BY new_flow_events.flow_id)
```

Comment:
I believe this should work well enough for historical attributions, because although the VPN's top-of-funnel `fxa_rp_button - view` events before 2021-12-08 wouldn't match these conditions, if the user clicked the button I think the next event in the flow should have `product_id` and `plan_id` set, and we generally can't attribute just based on `fxa_rp_button - view` events alone anyway because those almost never have `user_id` set.

However, this does mean it will work differently than [`mozilla_vpn_derived,fxa_attribution_v1`](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/fxa_attribution_v1/query.sql), which includes all `fxa_rp_button - view` and `fxa_pay_*` events before 2021-12-08. And it wouldn't work for [`mozilla_vpn_derived.funnel_product_page_to_subscribed_v1`](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/funnel_product_page_to_subscribed_v1/query.sql) as is, where counting all the top-of-funnel `fxa_rp_button - view` events matters.

Some potential options:
1. Change this logic to include the 89 million "VPN-related" `fxa_rp_button - view` events between 2021-08-25 and 2021-12-08 [like `funnel_product_page_to_subscribed_v1` does](https://github.com/mozilla/bigquery-etl/blob/647ff690f79b0389d5891d7a46aa43b1bf16eca5/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/funnel_product_page_to_subscribed_v1/query.sql#L160-L185).
2. Change this logic to include all 185 million `fxa_rp_button - view` events between 2021-08-25 and 2021-12-08.
3. Change this logic to include all 1.6 billion `fxa_rp_button - view` events before 2021-12-08 [like `mozilla_vpn_derived.fxa_attribution_v1` does](https://github.com/mozilla/bigquery-etl/blob/647ff690f79b0389d5891d7a46aa43b1bf16eca5/sql/moz-fx-data-shared-prod/mozilla_vpn_derived/fxa_attribution_v1/query.sql#L55-L80).
4. Don't change this logic, and if we rebuild that VPN funnel ETL to use this as the source then we'd have it only include data from 2021-12-08 onward.

For comparison on those event quantities, `mozilla_vpn_derived.fxa_attribution_v1` has captured 16 million VPN-specific flows after 2021-12-08 thus far.

I'm inclined to go with option 4, but what do you think @relud?

Reply:
4 sounds good